src/libmongoc/tests/test-mongoc-counters.c

```diff
@@ -481,6 +471,189 @@ test_counters_streams_timeout (void)
+
+   if (pooled) {
+      // Note: no server API version ensures OP_QUERY for initial handshake.
+      pool = test_framework_client_pool_new_from_uri (uri, NULL);
+      test_framework_set_pool_ssl_opts (pool);
+      mongoc_client_pool_set_error_api (pool, MONGOC_ERROR_API_VERSION_2);
+      client = mongoc_client_pool_pop (pool);
+   } else {
+      // Note: no server API version ensures OP_QUERY for initial handshake.
+      client = test_framework_client_new_from_uri (uri, NULL);
+      test_framework_set_ssl_opts (client);
+      mongoc_client_set_error_api (client, MONGOC_ERROR_API_VERSION_2);
+   }
+
+   mongoc_counter_auth_success_reset ();
+   mongoc_counter_auth_failure_reset ();
+   mongoc_counter_op_egress_query_reset ();
+   mongoc_counter_op_egress_msg_reset ();
+
+   ASSERT_OR_PRINT (
+      mongoc_client_get_server_status (client, NULL, NULL, &error), error);
+
+   const int32_t auth_success = mongoc_counter_auth_success_count ();
+   const int32_t auth_failure = mongoc_counter_auth_failure_count ();
+   const int32_t sent_queries = mongoc_counter_op_egress_query_count ();
+   const int32_t sent_msgs = mongoc_counter_op_egress_msg_count ();
+
+   // Ensure we are not testing more than we intend.
+   ASSERT_WITH_MSG (auth_success == 1 && auth_failure == 0,
+                    "expected exactly one authentication attempt to succeed, "
+                    "but observed %" PRId32 " successes and %" PRId32
+                    " failures",
+                    auth_success,
+                    auth_failure);
+
+   // MongoDB Handshake Spec: Since MongoDB server 4.4, the initial handshake
+   // supports a new argument, `speculativeAuthenticate`, provided as a BSON
+   // document. Clients specifying this argument to hello or legacy hello will
+   // speculatively include the first command of an authentication handshake.
+   const bool has_speculative_auth = test_framework_get_server_version () >=
+                                     test_framework_str_to_version ("4.4.0");
+
+   // The number of expected OP_QUERY requests depends on pooling and the
+   // presence of the RTT monitor thread.
+   if (pooled) {
+      // RTT monitoring is also a 4.4+ feature alongside speculative
+      // authentication and affects the number of OP_QUERY requests.
+      if (has_speculative_auth) {
+         // OP_QUERY requests consists of:
+         //  - initial connection handshake by server monitor thread (x3)
+         //  - initial connection handshake by RTT monitor thread (x3)
+         //  - polling hello by RTT monitor thread (x3)
+         //  - initial connection handshake by new cluster node
+         ASSERT_WITH_MSG (
+            sent_queries == 10,
+            "expected exactly ten OP_QUERY requests, but observed %" PRId32
+            " requests",
+            sent_queries);
+      } else {
+         // OP_QUERY requests consists of:
+         //  - initial connection handshake by server monitor (x3)
```

Comment:
The server monitor threads are started on the first call to `mongoc_client_pool_pop`. The OP_QUERY messages sent by the server monitoring threads may not be guaranteed to have completed.
I suggest adding an `ASSERT_SOON` to assert this condition happens within some time interval. This could be similar to [OBSERVE_SOON](https://github.com/mongodb/mongo-c-driver/blob/46a010f28d3052b3721e316b0bdecce47de519f1/src/libmongoc/tests/test-mongoc-background-monitoring.c#L286-L301).